### PR TITLE
Improve the application selection dropdown under Application audience role creation

### PR DESCRIPTION
**Pull request by dasuni-30**
### Purpose
Improve the application selection dropdown under Application audience role creation.

Currently, we can select applications with `organization` audience when creating an `application audience role` and this displays an error after creation. This PR disables showing the applications with `organization` audience when selecting an application for `application audience role` creation and provides an option to navigate to change the audience.

Screenshot 2023-11-01 at 11 02 34

### Related Issues
- https://github.com/wso2/product-is/issues/17351

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/roles/components/wizard-updated/role-basics.tsx [hunk 1]
```diff
@@ -93,12 +93,18 @@ export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicPr
...
+    const ROLES_TAB_INDEX: number = 5;
+
+
     const {
         data: applicationList,
         isLoading: isApplicationListFetchRequestLoading,
         error: applicationListFetchRequestError,
         mutate: mutateApplicationListFetchRequest
-    } = useApplicationList("clientId", null, null, applicationSearchQuery);
+    } = useApplicationList("clientId,associatedRoles.allowedAudience", null, null, applicationSearchQuery);
```
Comment (Rashmini) [inline apps/console/src/features/roles/components/wizard-updated/role-basics.tsx:107]:
```suggestion
    } = useApplicationList("clientId, associatedRoles.allowedAudience", null, null, applicationSearchQuery);
```

**Review by Rashmini — DISMISSED** [review]

File: apps/console/src/features/roles/components/wizard-updated/role-basics.tsx (hunk 1 above)
Comment (dasuni-30) [reply apps/console/src/features/roles/components/wizard-updated/role-basics.tsx:107]: These are the attributes we are passing in the API request URL. Adding space here makes the request invalid.

**Comment by Achintha444** [comment]
Add a full stop(.) after `Support organization-scoped roles`.

```
Support organization-scoped roles. Change the Audience
```

File: apps/console/src/features/roles/components/wizard-updated/role-basics.tsx [hunk 2]
```diff
@@ -204,6 +225,15 @@ export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicPr
...
 
+    /**
+     * Navigate to the Applications Edit page.
+     */
+    const navigateToApplicationEdit = (appId: string) => 
+        history.push({
+            pathname: AppConstants.getPaths().get("APPLICATION_SIGN_IN_METHOD_EDIT")
+                .replace(":id", appId).replace(":tabName", `#tab=${ ROLES_TAB_INDEX }`)
+        });
+
```
Comment (Achintha444) [inline apps/console/src/features/roles/components/wizard-updated/role-basics.tsx:236]: Ideally we should change the application audience from this interface it self. But for right now I think its okay.

Comment (dasuni-30) [reply apps/console/src/features/roles/components/wizard-updated/role-basics.tsx:236]: Ack

**Comment by dasuni-30** [comment]
> Add a full stop(.) after `Support organization-scoped roles`.
> 
> ```
> Support organization-scoped roles. Change the Audience
> ```

Improved with cb509f3bcc88871019cd92ac4c66a20eec154693

**Review by Rashmini — APPROVED** [review]

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**